### ISSUE TITLE
Creating a form *sometimes* doesn't create its submissions table in database

### ISSUE THREAD

**Issue by AlejandroDenche** [mautic/mautic#14223]
### Mautic Series

5.1.x series

### Mautic installed version

5.1.1

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

8.2.24

### What happened?

A week ago I upgraded from Mautic 4.4.x to Mautic 5.1.1. Everything seems to be working fine, until yesterday. The marketing team reported me an issue when users can't submit a form.

After checking Logs (attached as First Form Logs) I noticed there's no results table in DB. After some search, I found in Mautic's forum another user reporting similar issue (but not same), and the only thing someone said was to check for missing Migrations after upgrade, but no one provided a solution.  

I did that (doctrine:migrations:migrate) but everything seems ok. After running this, the table is still missing.

![Captura de pantalla 2024-10-25 a las 9 28 22](https://github.com/user-attachments/assets/c13aab3b-2af8-4b4f-a053-56fff74cf421)

So I decided to create a new Form. I did it from scratch (not clone) but with the same fields. When I clicked on save button, I got a 500 error. I checked logs and I got the Second Form Log. As same as first form, there's no submissions table for this Form.

For last, I created another form, but in this case I created the fields 1 by 1. In this way, everything worked fine, and now i have the form working.

### How can we reproduce this issue?

Step 1: Create a form
Step 2: Sometimes, the described error occurs

### Relevant log output

```shell
First Form Log: 

[2024-10-25T06:16:49.427483+00:00] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\TableNotFoundException: "An exception occurred while executing a query: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'db_name.form_results_20_solicitud' doesn't exist" at <mautic path>/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 49 {"exception":"[object] (Doctrine\\DBAL\\Exception\\TableNotFoundException(code: 1146): An exception occurred while executing a query: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'db_name.form_results_20_solicitud' doesn't exist at <mautic path>/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:49)\n[previous exception] [object] (Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1146): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'db_name.form_results_20_solicitud' doesn't exist at <mautic path>/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28)\n[previous exception] [object] (PDOException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'db_name.form_results_20_solicitud' doesn't exist at <mautic path>/vendor/doctrine/dbal/src/Driver/PDO/Connection.php:71)"} {"hostname":"<server_name>","pid":30019}


Second Form Log:

[2024-10-25T07:02:14.628988+00:00] mautic.CRITICAL: Uncaught PHP Exception Twig\Error\RuntimeError: "Key "showAfterXSubmissions" for sequence/mapping with keys "label, showLabel, properties, alias, labelAttributes, inputAttributes, containerAttributes, parent, type, formId, _token, buttons, id" does not exist." at <mautic_path>/app/bundles/FormBundle/Resources/views/Builder/_field_wrapper.html.twig line 39 {"exception":"[object] (Twig\\Error\\RuntimeError(code: 0): Key \"showAfterXSubmissions\" for sequence/mapping with keys \"label, showLabel, properties, alias, labelAttributes, inputAttributes, containerAttributes, parent, type, formId, _token, buttons, id\" does not exist. at <mautic_path>/app/bundles/FormBundle/Resources/views/Builder/_field_wrapper.html.twig:39)"} {"hostname":"<server_name>","pid":21111}
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by RCheesley** [comment]
Hi there,

I found a relevant bugfix here: https://github.com/mautic/mautic/pull/12220 relating to https://github.com/mautic/mautic/issues/12044 which should have fixed that issue about `showAfterXSubmissions` which was merged in 5.0-alpha but I can't find anything about the other issue. 

The second error should not be happening if your codebase is up to date with 5.1.1 - perhaps might be worth checking if you have that code from the bugfix in your instance?

Do you know if there were any issues when you updated from 4 to 5? It might be that there was an issue with the migration, perhaps?

**Comment by AlejandroDenche** [comment]
I've checked manually, and yes, that file affected by the PR is in my installation, and even with some extra lines (probably from another PR)

The only issue I noticed is this one. In the last month i've updated about 10 Mautic installations to 5.x, and there's no issue apart from this one. This issue only happened twice. The first time I didn't gave it so much importance since the form didn't have any submissions, so I decided to recreate the form manually and let it go.

In the next days I'll update a couple of Mautic installations, so I'll be watching anything strange. Will check every function of the installation.

Is there something that you recommend me to be aware of?

Thanks for your time!

**Comment by RCheesley** [comment]
Thanks @AlejandroDenche and sorry for the late response on my side. Seems weird that only one instance is affected. Could you just check if either have showAfterXSubmissions set to 0? We have another issue open about that and I wonder if it might perhaps be connected.

**Comment by RCheesley** [comment]
@AlejandroDenche I'm going to go ahead and close this owing to a lack of response but feel free to request a re-open if you're able to respond in the future.
